new/types.ts

```diff
@@ -114,8 +114,41 @@ export type Env = {
   /** affected messages */
   rawCommitMessages?: MessageRaw[];
   commitMessages?: Message[];
-  /** list of package.json files in the monorepo */
-  packageJsonPaths?: string[];
+  /** collection of affected packages for this release */
+  packages?: Record<string, Package>;
+};
+
+export type RelatedPackages = {
+  dependencies: string[];
+  devDependencies: string[];
+  peerDependencies: string[];
+};
+
+export type Package = {
+  /** Commit Messages stating what changed on this package (strictly) */
+  messages: Message[];
+  /** Commit Messages to be used for each (internal) dependency being released */
+  relatedMessages: Message[];
+  /**
+   * Collection of other (internal) Packages that depends on this
```

Comment:
I think the comments must be the other way round.

Probably, we should already make this clear in the variable names to not think about it in the future:

- `thisDependsOn`
- `packagesDependingOnThis`

Reply:
I aimed for a word(s) that reflect what their "dependency" relationship.

> The circumlocutions “A depends on B” and “B is depended on by A” are much more common in general use than “A is the depender, B is the ' dependee ”.

https://wikidiff.com/dependee/dependent

Reply:
But I admit it doesn't feel so intuitive and I'm open to change them.

what about a shorter version, like `dependsOn`:

`packages['@xingternal/button'].dependsOn['dependencies' | 'devDependencies' | 'peerDependencies']` instead of `packages['@xingternal/button'].thisDependsOn[...]`

As for `packagesDependingOnThis`, it's more intuitive, but can you think a shorter term?

Reply:
`dependsOn` makes absolute sense. I only added this because of the other variable. I couldn't think of a shorter term for that (but also wouldn't mind the long name if it saves brainpower in the future).